src/common/transformations/src/transformations/transpose_sinking/ts_utils.cpp

```diff
@@ -430,15 +428,47 @@ bool HasSameOutputTransposeNodes(const Output<Node>& output) {
     return HasSameOutputTransposeNodes(output.get_node_shared_ptr());
 }
 
-void RemoveSingleOutputConsumers(const NodePtr& node) {
+bool RemoveTransposeConsumers(const NodePtr& node) {
+    std::map<size_t, std::vector<ov::op::v1::Transpose*>> out_idx_to_redundant_transposes;
```

Comment:
consider to use std::unordered_map since it has constant complexity instead of std::map

Reply:
done